### PR TITLE
#149: Moved the API documentation to this repository.

### LINKED ISSUE #149: Move the API documentation from the main adapter repository
## Problem

The API documentation belongs here more than the main adapter, so we should move it here and link it from the main repository.

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #149

### CODE REVIEW

File: doc/development/api/scalar_functions_api.md [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+    {
+      "columnNr": 1,
+      "name": "C2",
+      "tableName": "ALL_EXASOL_TYPES",
+      "type": "column"
+    }
+  ]
+}
+```
+
```
Comment (exanm) [inline doc/development/api/scalar_functions_api.md:208]:
````suggestion

### Arithmetic Operators

Arithmetic operators use the common scalar function API, but have special names:

| Function Name | Operation     |
|---------------|---------------|
| NEG           | `-exp1`       |
| ADD           | `exp1 + exp2` |
| SUB           | `exp1 - exp2` |
| MULT          | `exp1 * exp2` |
| FLOAT_DIV     | `exp1 / exp2` |

```json
{
    "type": "function_scalar",
    "numArgs": 2,
    "name": "ADD",
    "infix": true,
    "arguments": [
    {
        ...
    },
    {
        ...
    }
    ]
}
```

````

File: doc/development/api/scalar_functions_api.md [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+    "arguments": [
+    {
+        ...
+    },
+    {
+        ...
+    }
+    ]
+}
+```
```
Comment (exanm) [inline doc/development/api/scalar_functions_api.md:412]:
```suggestion
```

See above.

File: doc/development/api/scalar_functions_api.md [hunk 3]
```diff
@@ -0,0 +1,439 @@
...
+## Exasol Scalar Functions List
+
+### Functions With a Common API
+
+The majority of the functions supported by the Exasol database shares a common API syntax depending on the number of function arguments:
+
+* [Functions without arguments](#functions-without-arguments);
+* [Functions with a single argument](#functions-with-a-single-argument);
+* [Functions with two or more arguments](#functions-with-two-or-more-arguments);
+* [Functions with a variable number of arguments](#functions-with-a-variable-number-of-arguments).
```
Comment (exanm) [inline doc/development/api/scalar_functions_api.md:14]:
```suggestion
* [Functions with a variable number of arguments](#functions-with-a-variable-number-of-arguments);
```
See below.

File: doc/development/api/scalar_functions_api.md [hunk 4]
```diff
@@ -0,0 +1,439 @@
...
+  Here belong a few functions, some of them are treated as a function with variable number of arguments despite the fact that actually the number of arguments is limited:
+  BIT_TO_NUM, CONCAT, DUMP, GREATEST, HASH_*, INSERT, INSTR, LEAST, LOCATE, REGEXP_INSTR, REGEXP_REPLACE, REGEXP_SUBSTR.
+
```
Comment (exanm) [inline doc/development/api/scalar_functions_api.md:17]:
```suggestion
* [Arithmetic operators](#arithmetic-operators).

```

**Review by exanm — CHANGES_REQUESTED** [review]
I also reviewed the aggregated functions.

We should add some general notice that analytic functions are not included.

File: doc/development/api/virtual_schema_api.md [hunk 5]
```diff
@@ -0,0 +1,1210 @@
...
+
+The same can be used for a predicate type `predicate_is_not_json` (requires predicate capability `IS_NOT_JSON`).
+
+### Scalar Functions
+
+Refer to the [Exasol Scalar Functions API Documentation](scalar_functions_api.md)
+
+### Aggregate Functions
+
+Consistent with scalar functions. To be detailed: `star-operator`, `distinct`, ...
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply doc/development/api/virtual_schema_api.md:1037]: I have no idea what it's supposed to mean

File: doc/development/api/virtual_schema_api.md [hunk 6]
```diff
@@ -0,0 +1,1238 @@
...
+    {
+        ...
+    }
+    ]
+}
+```
+##### COUNT(*)
+
+`COUNT(*)`
+ (requires set-function capability `COUNT` and `COUNT_STAR`)
```
Comment (exanm) [inline doc/development/api/virtual_schema_api.md:1119]:
```suggestion
 (Requires set-function capability `COUNT_STAR`. Please notice, that the set-function capability `COUNT` is not required in this case.)
```

File: doc/development/api/virtual_schema_api.md [hunk 7]
```diff
@@ -0,0 +1,1238 @@
...
+{
+    "type": "function_aggregate",
+    "name": "COUNT"
+}
+```
+
+##### COUNT(DISTINCT exp)
+
+`COUNT(DISTINCT exp)`
+ (requires set-function capability `COUNT` and `COUNT_DISTINCT`)
```
Comment (exanm) [inline doc/development/api/virtual_schema_api.md:1131]:
```suggestion
 (requires set-function capabilities `COUNT` and `COUNT_DISTINCT`)
```

File: doc/development/api/virtual_schema_api.md [hunk 8]
```diff
@@ -0,0 +1,1238 @@
...
+    "arguments": [
+    {
+        ...
+    },
+    {
+        ...
+    }
+    ]
+}
+```
```
Comment (exanm) [inline doc/development/api/virtual_schema_api.md:1165]:
```suggestion
```
Not implemented, yet.
```

File: doc/development/api/virtual_schema_api.md [hunk 9]
```diff
@@ -0,0 +1,1238 @@
...
+
+##### AVG(DISTINCT exp)
+
+`AVG(DISTINCT exp)`
+ (requires set-function capability `AVG` and `AVG_DISTINCT`)
```
Comment (exanm) [inline doc/development/api/virtual_schema_api.md:1187]:
```suggestion
 (requires set-function capabilities `AVG` and `AVG_DISTINCT`)
```

File: doc/development/api/virtual_schema_api.md [hunk 10]
```diff
@@ -0,0 +1,1238 @@
...
+
+| Function Name       | API Mapping Link                                   |
+|---------------------|----------------------------------------------------|
+| AVG(exp)            | [AVG(exp) function](#avgexp)                       |
+| AVG(DISTINCT exp)   | [AVG(DISTINCT exp) function](#avgdistinct-exp)     |
+| COUNT(*)            | [COUNT(*) function](#count)                        |
+| COUNT(exp)          | [COUNT(exp) function](#countexp)                   |
+| COUNT(DISTINCT exp) | [COUNT(DISTINCT exp) function](#countdistinct-exp) |
+| COUNT((exp1, exp2)) | [COUNT((exp1, exp2)) function](#countexp1-exp2)    |
+| GROUP_CONCAT        | [GROUP_CONCAT function](#group_concat)             |
```
Comment (exanm) [inline doc/development/api/virtual_schema_api.md:1084]:
```suggestion
| Function Name       | API Mapping Link                                   |
|---------------------|----------------------------------------------------|
| AVG(exp)            | [AVG(exp) function](#avgexp)                       |
| AVG(DISTINCT exp)   | [AVG(DISTINCT exp) function](#avgdistinct-exp)     |
| COUNT(*)            | [COUNT(*) function](#count)                        |
| COUNT(exp)          | [COUNT(exp) function](#countexp)                   |
| COUNT(DISTINCT exp) | [COUNT(DISTINCT exp) function](#countdistinct-exp) |
| GROUP_CONCAT        | [GROUP_CONCAT function](#group_concat)             |
```

File: doc/development/api/virtual_schema_api.md [hunk 11]
```diff
@@ -0,0 +1,1238 @@
...
+|-----------------|-----------------------------------------|
+| ANY             | The API uses the SOME function.         |
+| CORR            | Not included in the API.                |
+| COVAR_POP       | Not included in the API.                |
+| COVAR_SAMP      | Not included in the API.                |
+| GROUPING        | Not included in the API.                |
+| LISTAGG         | The API uses the GROUP_CONCAT function. |
+| PERCENTILE_CONT | Not included in the API.                |
+| PERCENTILE_DISC | Not included in the API.                |
+| REGR_*          | Not included in the API.                |
```
Comment (exanm) [inline doc/development/api/virtual_schema_api.md:1098]:
```suggestion
| Function Name       | Comment                                 |
|---------------------|-----------------------------------------|
| ANY                 | The API uses the SOME function.         |
| CORR                | Not included in the API.                |
| COUNT((exp1, exp2)) | Not included in the API.                |
| COVAR_POP           | Not included in the API.                |
| COVAR_SAMP          | Not included in the API.                |
| GROUPING            | Not included in the API.                |
| LISTAGG             | The API uses the GROUP_CONCAT function. |
| PERCENTILE_CONT     | Not included in the API.                |
| PERCENTILE_DISC     | Not included in the API.                |
| REGR_*              | Not included in the API.                |
```

File: doc/development/api/virtual_schema_api.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (exanm) [reply doc/development/api/virtual_schema_api.md:1039]:
```suggestion
```

**Review by exanm — APPROVED** [review]
See suggested changes.

**Review by exanm — APPROVED** [review]

**Pull request merged**